README.md

```diff
@@ -1,5 +1,7 @@
 [![Build Status](https://travis-ci.com/paypal/NNAnalytics.svg?branch=master)](https://travis-ci.com/paypal/NNAnalytics)
 [![License](http://img.shields.io/:license-Apache%202-blue.svg)](http://www.apache.org/licenses/LICENSE-2.0.txt)
+[![Codacy Badge](https://api.codacy.com/project/badge/Grade/d47519e1762d4a74ae5720b05450a667)](https://www.codacy.com/app/Dee-Pac/NNAnalytics?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=paypal/NNAnalytics&amp;utm_campaign=Badge_Grade)
```

Comment:
Does the username have to be there for the badge? The full app should be `/paypal/NNAnalytics`.

Reply:
@zero45 Yes, I believe so. I don't see an option to make it not user specific.

We could go with this for now and revisit later if there are other options. I've not seen any issue with similar approach in Gimel since the time we open sourced.

Reply:
While usually I would not personally care, I do know that Apache tries very hard to avoid tagging any one thing with someone's username / personal identity. I doubt it would be a major issue for now but becoming an ASF project is a goal so we need to treat this seriously. 

It may come to that we need to have just the badge and no URL + username at some point.
Just a heads up.

+1 though. Let's move on.
I will prefer to keep code coverage done by CodeCov; incase that was a thought.

Reply:
@zero45 

Let me know the option you prefer.

1. I can remove the codacy right away, and we can be future proof.

2. Keep the badge and determine at a later point how to make it user ID agnostic. We can always delete the reference if not achievable. We choose this route with Gimel & I was in same dilemma at gimel oss - we looked at squbs, and decided to take option#2. 
Codacy could be mostly used for coding patterns, security coverages, etc - especially on PRs - you will get automated notification on code quality. 

LMK. 

BTW - I don't think CodeCov does the same thing as Codacy, or vice versa  : (apples - oranges comparison)

Reply:
I think I am OK with taking number 2. Apologies for the indecision at first I mostly just wanted to get my thoughts out.

I was aware that CodeCov and Codacy are not equal -- however Codacy does allow you to report code coverage as well. All I was saying is to keep code coverage done by CodeCov. Clearly Codacy does much more than code coverage and has value.

+1 on option 2.

Reply:
@zero45 Is this ready for merge then?

Reply:
FYI https://shields.io/ shows 4 shield options for codacy : code quality vs code coverage (for master vs branch).

Reply:
@r39132 The coverage shows as X (nothing) when I look at the badge for coverage. 
Do you still like to have that?

Reply:
@Dee-Pac Hmn... so maybe we stick with CodeCov for coverage but use the code quality shield from Codacy? Are you you suggesting that we use the general Codacy badge?

Reply:
@r39132 Right - CodeCov for coverage / general code quality via codacy. 

Idea is we could have folks look at codacy dashboard and improve aspects like these pulled from dashboard.

https://app.codacy.com/project/Dee-Pac/NNAnalytics/dashboard

```
Security |   | 8
Error Prone |   | 116
Code Style |   | 443
Compatibility |   | 32
Unused Code |   | 8
Performance |   | 0
```